pkg/cmd/roachtest/roachtestutil/clusterupgrade/clusterupgrade.go

```diff
@@ -368,19 +368,26 @@ func RestartNodesWithNewBinary(
 	rand.Shuffle(len(nodes), func(i, j int) {
 		nodes[i], nodes[j] = nodes[j], nodes[i]
 	})
+
+	// Stop the cockroach process gracefully in order to drain it properly.
+	// This makes the upgrade closer to how users do it in production, but
+	// it's also needed to eliminate flakiness. In particular, this will
+	// make sure that DistSQL draining information is dissipated through
```

Comment:
Nit: maybe `communicated`? `dissipated` is synonymous with `evaporated` :)

Reply:
Updated. I just moved this comment in the PR, it has been around for a while I believe.